src/Components/Components/src/Rendering/RenderBatchBuilder.cs

```diff
@@ -18,7 +18,7 @@ internal class RenderBatchBuilder : IDisposable
         // Primary result data
         public ArrayBuilder<RenderTreeDiff> UpdatedComponentDiffs { get; } = new ArrayBuilder<RenderTreeDiff>();
         public ArrayBuilder<int> DisposedComponentIds { get; } = new ArrayBuilder<int>();
```

Comment:
Should component ids be on the same boat? I can imagine that if we don't want to ever wrap around an event, it might be even more important for components, as it might be the case that 1 event triggers multiple component instantiations.

Thoughts?

I know the struct layout might get more complicated if we add 4 more bytes. But I think its a matter of being better safe than sorry.

Reply:
I’m open to it, but it will be quite a lot more disruptive than the event handler ID change, as component IDs are used in so many more places. 

To run out of IDs, you’d need to do something like instantiate 1000 components per second continuously for 25 days (in one circuit). I think we could say that is impractical for lots of reasons.

One cheaper option would just be to change any public API for component IDs to be ulong, which leaves us the option of changing the underlying storage in the future if we want.